src/extension/geopkg-output/src/main/java/org/geoserver/geopkg/wfs/GeoPackageGetFeatureOutputFormat.java

```diff
@@ -71,14 +78,42 @@ protected String getExtension(FeatureCollectionResponse response) {
         return EXTENSION;
     }
 
+    /**
+     * This is a smarter version of File.createTempFile().
+     *
+     * <p>If the CUSTOM_TEMP_DIR_PROPERTY property is set, then use that as the temp directory.
+     * Otherwise (unset), use the standard File.createTempFile() temp directory.
+     *
+     * <p>If the CUSTOM_TEMP_DIR_PROPERTY directory doesn't exist (or isnt a directory), then an
+     * exception is thrown
+     *
+     * @param prefix
+     * @param suffix
+     * @return
+     */
+    File createTempFile(String prefix, String suffix) throws IOException {
+        String customTempDir = GeoServerExtensions.getProperty(CUSTOM_TEMP_DIR_PROPERTY);
+        if (!StringUtils.hasText(customTempDir)) {
+            return File.createTempFile("geopkg", ".tmp.gpkg");
+        }
+        File tempDir = new File(customTempDir);
+        if (!tempDir.exists() || !tempDir.isDirectory()) {
+            throw new IOException(
```

Comment:
```suggestion
            throw new FileNotFoundException(
```

Will need to adjust import also